compiler/luci/service/src/CircleShapeInferenceRule.test.cpp

```diff
@@ -48,6 +49,54 @@ bool shape_pass(loco::Graph *g)
   return loco::apply(&rules).to(g);
 }
 
+bool is_same_shape(luci::CircleNode *node, loco::TensorShape shape)
+{
+  if (node->shape_status() != luci::ShapeStatus::VALID)
+    return false;
+
+  if (node->rank() != shape.rank())
+    return false;
+
+  for (uint32_t i = 0; i < node->rank(); ++i)
+  {
+    if (node->dim(i).known() != shape.dim(i).known())
+      return false;
+
+    if (node->dim(i).value() != shape.dim(i).value())
+      return false;
+  }
+
+  return true;
+}
+
+// NOTE This function imitates CircleShapeInferencePass but little bit different.
+//      In CircleShapeInferencePass, DeadNodeQueryService is used to get alive nodes,
+//      which are not detected with postorder_traversal.
+//      However, it is not considered in this function because this is just for testing
+//      inference rule itself, not for inference pass.
+bool circle_shape_pass(loco::Graph *g)
```

Comment:
Where is this function called from?

Reply:
This function will substitute `bool shape_pass(loco::Graph *g)` function to use `luci` shape at another PR.

Reply:
If you don't know why I asked,
- new introduced codes are not tested by our CI
- there is no DRAFT link to understand the usage of this code

The code seems OK but the whole usage is unknown to me, so I can not agree with this chage by looking at this PR.

Reply:
Sorry for bothering you..

I created [Draft PR](https://github.com/Samsung/ONE/pull/5756/commits/c8d3066f0a2440b659579abef7ab6b7bfeb56bf2).
If I apply all of the changes in this PR at once, I thought that code diff would be very large and hard to review.
So I divided the changes as
- Introduce new function
- Enable new function and substitute related functions
I hope it helps your review...